src/app/interaction-model/invoke_responder.h

```diff
@@ -0,0 +1,108 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/data-model/WrappedStructEncoder.h>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+namespace InteractionModel {
+
+/// Aids in writing invoke replies:
+///   - Has access to the TLV writer of the response data content
```

Comment:
Will update to reference `ReplyEncoder`. I believe originally I wanted to have direct TLVWriter however then I saw that our typical repluies use a wrappedstructencoder.

I expect this may end up changing as we actually try to implement it. Overall I need something that I can cleanly say `Reply().Send(my_repy_struct)`